src/amo/components/UserProfile/index.js

```diff
@@ -0,0 +1,170 @@
+/* @flow */
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import ReportUserAbuse from 'amo/components/ReportUserAbuse';
+import { fetchUserAccount } from 'amo/reducers/users';
+import { withErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import { sanitizeUserHTML } from 'core/utils';
+import Card from 'ui/components/Card';
+import DefinitionList, { Definition } from 'ui/components/DefinitionList';
+import LoadingText from 'ui/components/LoadingText';
+import Rating from 'ui/components/Rating';
+import UserAvatar from 'ui/components/UserAvatar';
+import type { UsersStateType, UserType } from 'amo/reducers/users';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+
+type UserProfileParams = {| username: string |};
+
+type Props = {|
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  params: UserProfileParams,
+  user: UserType,
+|};
+
+export class UserProfileBase extends React.Component<Props> {
+  componentWillMount() {
+    const { dispatch, errorHandler, params, user } = this.props;
+
+    if (!user) {
+      dispatch(fetchUserAccount({
+        errorHandlerId: errorHandler.id,
+        username: params.username,
+      }));
+    }
+  }
+
+  componentWillReceiveProps({ params: newParams }: { params: UserProfileParams }) {
+    const { dispatch, errorHandler, params: oldParams } = this.props;
+
+    if (oldParams.username !== newParams.username) {
+      dispatch(fetchUserAccount({
+        errorHandlerId: errorHandler.id,
+        username: newParams.username,
+      }));
+    }
+  }
+
+  render() {
+    const {
+      errorHandler,
+      i18n,
+      params,
+      user,
+    } = this.props;
+
+    if (errorHandler.hasError()) {
+      log.warn('Captured API Error:', errorHandler.capturedError);
+
+      if (errorHandler.capturedError.responseStatusCode === 404) {
+        return <NotFound errorCode={errorHandler.capturedError.code} />;
+      }
+    }
+
+    const userProfileHeader = (
+      <React.Fragment>
+        <UserAvatar className="UserProfile-avatar" user={user} />
+
+        <h1 className="UserProfile-name">
+          {user ? user.displayName : <LoadingText />}
+        </h1>
+      </React.Fragment>
+    );
+    const userProfileTitle = i18n.sprintf(
+      i18n.gettext('User Profile for %(user)s'), {
+        user: user ? user.displayName : params.username,
+      }
+    );
+
+    return (
+      <div className="UserProfile">
+        <Helmet>
+          <title>{userProfileTitle}</title>
+        </Helmet>
+
+        {errorHandler.renderErrorIfPresent()}
```

Comment:
Since you're already checking for the error, the rendering logic should all happen in the same place to avoid confusion. It can look like this up above:

```js
let errorMessage;
if (errorHandler.hasError()) {
  if (errorHandler.capturedError.responseStatusCode === 404) {
    return <NotFound .../>;
  }

  errorMessage = errorHandler.renderError();
}
```
...
```js
return (
  <div className="UserProfile">
    {errorMessage}
  </div>
);
```

Otherwise it looks like you are forgetting to handle the error in the first if statement.

Reply:
I saw that in other components and I thought it seemed more verbose... but fair point about it not looking like an error is checked until then otherwise. I'll change it 👍